frontend/src/metabase/selectors/metadata.js

```diff
@@ -123,8 +124,36 @@ export const getSegments = createSelector(
 
 // MISC
 
-export const getParameterFieldValues = (state, props) => {
-    return getFieldValues(getIn(state, ["metadata", "fields", props.parameter.field_id]));
+// Returns a dictionary of field id:s mapped to matching field values
+const getParameterFieldValuesByFieldId = (state, props) => _.chain(getFields(state))
+    .pick(getFields(state), ...props.parameter.field_ids)
+    .mapObject(getFieldValues)
+    .value()
+
+// Check if the lengths of field value arrays equal for each field
+// TODO: Why can't we use plain shallowEqual, i.e. why the field value arrays change very often?
+const createFieldValuesEqualSelector = createSelectorCreator(defaultMemoize, (a, b) => {
```

Comment:
what does "defaultMemoize" mean?

Reply:
It's the default memoization strategy of `reselect` which means that always when selector arguments change, the selector value is recalculated (as opposed to maintaining a cache of old selector argument <-> selector value combos and using cached values if possible, for instance).

Sadly the terminology of `reselect` isn't often very descriptive.

Reply:
I should probably add an explanatory comment to the code above, in minimum by adding a link to https://github.com/reactjs/reselect#customize-equalitycheck-for-defaultmemoize